### DISCUSSION TITLE
How to send velocity setpoints continuously in a loop for motion commander?

### DISCUSSION THREAD

**Discussion by Vrinda73** [bitcraze/discussions#461] [Q&A]
Hello! 
This might be a silly question. I apologize for it. Please consider me as a beginner.

Question: I want to send velocity setpoints to CF using motion commander. However, my setpoints will change every time instance as per my algorithm settings. I was wondering how I can design the duration/ time.sleep value  in each iteration.

Thank you so much!

**Comment by krichardsson** [comment]
I don't think there is a good way to do what yo want with the Motion Commander, it sounds like you maybe should not use it in this case. 

The MotionCommander is really just a set of methods that sends simple velocity setpoints, and some threading to make it easy to use. If you check out the code in the python lib you will find that what you need to do is mainly to call [`cf.commander.send_hover_setpoint()`](https://github.com/bitcraze/crazyflie-lib-python/blob/1c07945e7f93587d9e5afd860258aa304b61555d/cflib/positioning/motion_commander.py#L465)

You can find more types of setpoints in the [`commander`](https://github.com/bitcraze/crazyflie-lib-python/blob/master/cflib/crazyflie/commander.py) if you want to check them out

**Reply by Vrinda73** [reply]
Thank you very much Kristoffer! I will try what you suggested. Regarding time.sleep(?), should I design it by trial and error ?

**Reply by krichardsson** [reply]
If you are using `send_hover_setpoint(self, vx, vy, yawrate, zdistance)` vx and vy are the desired velocity in m/s. A good estimate for when you will reach the destination is to divide distance by velocity, it is not completely true as the CF is not moving at a constant velocity, but it will be pretty good.

One thing I forgot to mention is that you must send setpoints continuously, I think at least 4 times per second. There is a safety mechanism in the Crazyflie that will stop the motors otherwise (assuming it lost connection with the radio and we better stop). This is why there is a [thread in the MotionCommander](https://github.com/bitcraze/crazyflie-lib-python/blob/e5d2cd0eb407365f09f4197a5ecfe1d0df428468/cflib/positioning/motion_commander.py#L414-L479) that sends the setpoint 5 times/s. When I think about it, it might be convenient for you to reuse the thread? Just start it and call `set_vel_setpoint()` when you want it to send a different setpoint.
